bundles/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/handler/SolarEdgeGenericHandler.java

```diff
@@ -0,0 +1,238 @@
+import org.openhab.binding.solaredge.internal.command.AggregateDataUpdatePublicApi;
+import org.openhab.binding.solaredge.internal.command.LiveDataUpdateMeterless;
+import org.openhab.binding.solaredge.internal.command.LiveDataUpdatePrivateApi;
+import org.openhab.binding.solaredge.internal.command.LiveDataUpdatePublicApi;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.openhab.binding.solaredge.internal.config.SolarEdgeConfiguration;
+import org.openhab.binding.solaredge.internal.connector.CommunicationStatus;
+import org.openhab.binding.solaredge.internal.connector.WebInterface;
+import org.openhab.binding.solaredge.internal.model.AggregatePeriod;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelGroupUID;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolarEdgeGenericHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Friese - initial contribution
+ */
+@NonNullByDefault
+public class SolarEdgeGenericHandler extends BaseThingHandler implements SolarEdgeHandler, AtomicReferenceTrait {
+    private final Logger logger = LoggerFactory.getLogger(SolarEdgeGenericHandler.class);
+
+    private static final long LIVE_POLLING_INITIAL_DELAY = 1;
+    private static final long AGGREGATE_POLLING_INITIAL_DELAY = 2;
+
+    /**
+     * Interface object for querying the Solaredge web interface
+     */
+    private WebInterface webInterface;
+
+    /**
+     * Schedule for polling live data
+     */
+    private final AtomicReference<@Nullable Future<?>> liveDataPollingJobReference;
+
+    /**
+     * Schedule for polling aggregate data
+     */
+    private final AtomicReference<@Nullable Future<?>> aggregateDataPollingJobReference;
+
+    public SolarEdgeGenericHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.webInterface = new WebInterface(scheduler, this, httpClient);
+        this.liveDataPollingJobReference = new AtomicReference<>(null);
+        this.aggregateDataPollingJobReference = new AtomicReference<>(null);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("command for {}: {}", channelUID, command);
+        // write access is not supported.
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("About to initialize SolarEdge");
+        SolarEdgeConfiguration config = getConfiguration();
+        logger.debug("Solaredge initialized with configuration: {}", config);
```

Comment:
```suggestion
        logger.debug("SolarEdge initialized with configuration: {}", config);
```

(consistency with line 93)